api/public/cerbos/authzen/v1/authzen.proto

```diff
@@ -0,0 +1,251 @@
+  // Action name, e.g. "view".
+  string name = 2 [
+    (buf.validate.field).string.min_len = 1,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "Action name."
+      example: "\"view:public\""
+    }
+  ];
+}
+
+// Resource represents the AuthZEN resource entity.
+message Resource {
+  option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_schema) = {
+    json_schema: {description: "AuthZEN Resource"}
+  };
+  // Arbitrary key-value properties attached to the resource.
+  google.protobuf.Struct properties = 1 [(grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {description: "Arbitrary key-value properties attached to the resource."}];
+  // Resource type, e.g. "album:object".
+  string type = 2 [
+    (buf.validate.field).string.min_len = 1,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "Resource type."
+      example: "\"leave_request\""
+    }
+  ];
+  // Resource identifier.
+  string id = 3 [
+    (buf.validate.field).string.min_len = 1,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "Resource identifier."
+      example: "\"XX125\""
+    }
+  ];
+}
+
+// Tuple encapsulates a subject-action-resource tuple with optional context.
+message Tuple {
+  option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_schema) = {
+    json_schema: {description: "Subject-Action-Resource tuple with optional context"}
+  };
+  Subject subject = 1;
+  Action action = 2;
+  Resource resource = 3;
+  // Evaluation context passed alongside the tuple.
+  google.protobuf.Struct context = 4 [(grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {description: "Contextual attributes for this evaluation."}];
+}
+
+// EvaluationRequest is the AuthZEN evaluation request.
+// It supports a single tuple defined at the top level, or a batch via 'evaluations'.
+message AccessEvaluationRequest {
+  option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_schema) = {
+    json_schema: {description: "AuthZEN Access Evaluation request"}
+  };
+  Subject subject = 1 [
+    (buf.validate.field).required = true,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {description: "Subject to evaluate."}
+  ];
+  Action action = 2 [
+    (buf.validate.field).required = true,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {description: "Action to evaluate."}
+  ];
+  Resource resource = 3 [
+    (buf.validate.field).required = true,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {description: "Resource to evaluate."}
+  ];
+  google.protobuf.Struct context = 4 [(grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {description: "Optional evaluation context."}];
+}
+
+// EvaluationsRequest is the batch request form for multiple tuples.
+message AccessEvaluationsRequest {
```

Comment:
This naming is bad because it's just a single letter difference between `AccessEvaluationRequest` and `AccessEvaluationsRequest`.